packages/fast-components-react-msft/src/action-toggle/action-toggle.tsx

```diff
@@ -0,0 +1,239 @@
+        }
+    }
+
+    /**
+     * Render Glyphs
+     */
+    public renderGlyph(): React.ReactNode {
+        if (this.state.selected) {
+            return this.renderSelectedGlyph();
+        } else {
+            return this.renderUnselectedGlyph();
+        }
+    }
+
+    public renderSelectedGlyph(): React.ReactNode {
+        if (typeof this.props.selectedGlyph === "function") {
+            return this.props.selectedGlyph(
+                get(this.props, "managedClasses.actionToggle_selectedGlyph")
+            );
+        }
+        return null;
+    }
+
+    public renderUnselectedGlyph(): React.ReactNode {
+        if (typeof this.props.unselectedGlyph === "function") {
+            return this.props.unselectedGlyph(
+                get(this.props, "managedClasses.actionToggle_unselectedGlyph")
+            );
+        }
+        return null;
+    }
+
+    /**
+     * Generates class names
+     */
+    protected generateClassNames(): string {
+        let classNames: string = get(this.props, "managedClasses.actionToggle") || "";
+
+        if (this.props.disabled) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.actionToggle__disabled"
+            )}`;
+        }
+
+        if (this.state.selected) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.actionToggle__selected"
+            )}`;
+        }
+
+        if (this.isSingleElement()) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.actionToggle__singleElement"
+            )}`;
+        }
+
+        switch (this.props.appearance) {
+            case ActionToggleAppearance.primary:
+                classNames = `${classNames} ${get(
+                    this.props,
+                    "managedClasses.actionToggle__primary"
+                )}`;
+                break;
+            case ActionToggleAppearance.lightweight:
+                classNames = `${classNames} ${get(
+                    this.props,
+                    "managedClasses.actionToggle__lightweight"
+                )}`;
+                break;
+            case ActionToggleAppearance.justified:
+                classNames = `${classNames} ${get(
+                    this.props,
+                    "managedClasses.actionToggle__justified"
+                )}`;
+                break;
+            case ActionToggleAppearance.outline:
+                classNames = `${classNames} ${get(
+                    this.props,
+                    "managedClasses.actionToggle__outline"
+                )}`;
+                break;
+        }
+
+        return super.generateClassNames(classNames);
+    }
+
+    /**
+     * Checks to see if the toggle is displaying both glyph and text or not
+     */
+    private isSingleElement(): boolean {
+        if (this.state.selected) {
+            if (
```

Comment:
not 100% sure but it might fix your complexity issue to do this instead of an if statement:

```
return this.state.selected
?  isNullOrWhiteSpace(this.props.selectedText) || isNullOrUndefined(this.props.selectedGlyph)
: isNullOrWhiteSpace(this.props.unselectedText) || isNullOrUndefined(this.props.unselectedGlyph)